applications/GeoMechanicsApplication/custom_elements/permeability_calculator.h

```diff
@@ -0,0 +1,89 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#pragma once
+
+#include "calculator.h"
+
+#include <custom_retention/retention_law.h>
+
+#include <utility>
+
+namespace Kratos
+{
+
+class PermeabilityCalculator : public Calculator
+{
+public:
+    class InputProvider
+    {
+    public:
+        InputProvider(std::function<const Properties&()>                         rElementProperties,
+                      std::function<const std::vector<RetentionLaw::Pointer>&()> rRetentionLaws,
+                      std::function<Vector()>                        rIntegrationCoefficients,
+                      std::function<Vector(const Variable<double>&)> rNodalValuesOfDtWaterPressure,
+                      std::function<Geometry<Node>::ShapeFunctionsGradientsType()> rShapeFunctionGradients)
```

Comment:
Since these function objects act like 'getters', I would suggest to express that in the function parameter names:
```suggestion
        InputProvider(std::function<const Properties&()>                         GetElementProperties,
                      std::function<const std::vector<RetentionLaw::Pointer>&()> GetRetentionLaws,
                      std::function<Vector()>                        GetIntegrationCoefficients,
                      std::function<Vector(const Variable<double>&)> GetNodalValuesOfDtWaterPressure,
                      std::function<Geometry<Node>::ShapeFunctionsGradientsType()> GetShapeFunctionGradients)
```

Since the function parameters are passed by value rather than by reference, we don't need the `r` prefix.

In this way, the function parameter names are also more in line with the names of the data members that will keep these function objects.

Reply:
Done